src/net45/WampSharp.WAMP1/WAMP1/V1/Rpc/Client/WampRpcClientHandler.cs

```diff
@@ -30,6 +30,27 @@ public WampRpcClientHandler(IWampServerProxyFactory<TMessage> serverProxyFactory
         {
             mFormatter = formatter;
             mServerProxy = serverProxyFactory.Create(new RpcWampClient(this), connection);
+
+            connection.ConnectionClosed += Connection_ConnectionClosed;
+            connection.ConnectionError += Connection_ConnectionError;
+        }
+
+        private void Connection_ConnectionError(object sender, WampConnectionErrorEventArgs e)
+        {
+            HandleConnectionIssue("Connection Error");
+        }
+
+        private void Connection_ConnectionClosed(object sender, EventArgs e)
+        {
+            HandleConnectionIssue("Connection Closed");
+        }
+
+        private void HandleConnectionIssue(string errorDesc)
+        {
+            foreach (var kvp in mCallIdToSubject)
+            {
+                ErrorArrived(kvp.Key, "http://api.wamp.ws/error#generic", errorDesc);
```

Comment:
This calls mCallIdToSubject.TryRemove while enumerating mCallIdToSubject it, I think it won't work.